### PR TITLE
feat: add gas-comparison PR bot

### LINKED ISSUE #163: dev: run gas snapshot compare python script in the CI and comment results in PR
```[tasklist]
### Tasks
- [ ] Create a CI job that runs on PR
- [ ] The job runs the compare python script
- [ ] Comments the result on the PR as a report
```

### PR DESCRIPTION
**Pull request by enitrat**
## Pull Request type

Please check the type of change your PR introduces:

- [ ] Bugfix
- [ ] Feature
- [ ] Code style update (formatting, renaming)
- [ ] Refactoring (no functional changes, no API changes)
- [ ] Build-related changes
- [ ] Documentation content changes
- [ ] Other (please describe):

## What is the current behavior?

Issue Number: Closes #163 

## What is the new behavior?

-
-
-

## Does this introduce a breaking change?

- [ ] Yes
- [x] No

## Added changes to [`CHANGELOG.md`](../CHANGELOG.md)

- [x] Yes
- [ ] No

### CODE REVIEW

File: .github/workflows/gas_reports.yml [hunk 1]
```diff
@@ -0,0 +1,44 @@
...
+
+      - name: Set up Python
+        uses: actions/setup-python@v2
+        with:
+          python-version: "3.x" # Specify the Python version required
+
+      - name: Set up Scarb
+        uses: software-mansion/setup-scarb@v1
+        with:
+          scarb-version: "nightly-2023-08-12"
```
Comment (Eikix) [inline .github/workflows/gas_reports.yml:24]: thought for later but chances are we'll forget to change all the places we mention `scarb-version` everytime we bump, is there a way to change it in one place & change it in all actions at once?

**Review by Eikix — COMMENTED** [review]
lgtm

File: .github/workflows/gas_reports.yml (hunk 1 above)
Comment (enitrat) [reply .github/workflows/gas_reports.yml:24]: just cmd-f replace

**Comment by enitrat** [comment]
@Eikix please approve

**Review by Eikix — APPROVED** [review]

**Pull request merged**